scvi/train/_trainrunner.py

```diff
@@ -71,12 +74,42 @@ def __call__(self):
         self.training_plan.n_obs_training = len(self.model.train_indices)
 
         self.trainer.fit(self.training_plan, self.data_splitter)
-        try:
-            self.model.history_ = self.trainer.logger.history
-        except AttributeError:
-            self.history_ = None
+        self._update_history()
 
         self.model.module.eval()
         self.model.is_trained_ = True
         self.model.to_device(self.device)
         self.model.trainer = self.trainer
+
+    def _update_history(self):
+        # model is being further trained
+        if self.model.history_ is not None:
+            # if not using the default logger (e.g., tensorboard)
+            if not isinstance(self.model.history_, dict):
+                warnings.warn(
+                    "Training history cannot be updated. Replacing old history with new history."
+                )
+                if hasattr(self.model, "history_"):
+                    self.model.history_ = self.trainer.logger.history
```

Comment:
Let's move the warning message into the if statement, so it only warns if it actually does replace the old history. Also, I'm pretty sure every model has a history_ attribute. It's in the init of BaseModelClass.